packages/core/src/tools/limitModification.spec.ts

```diff
@@ -47,18 +48,33 @@ describe('limitModification', () => {
     })
   })
 
-  it('should not allow non string value on modifiable field', () => {
-    const object = { foo: { bar: 'bar' }, qux: 'qux' }
+  it('should not allow changing the type of the value on modifiable field', () => {
+    const object = {
+      string_to_undefined: 'bar',
+      string_to_number: 'qux',
+      null_to_object: null,
+      object_to_null: {},
+      array_to_object: [],
+      object_to_array: {},
+    }
     const modifier = (candidate: any) => {
-      candidate.foo.bar = undefined
-      candidate.qux = 1234
+      candidate.string_to_undefined = undefined
+      candidate.string_to_number = 1234
+      candidate.null_to_object = {}
+      candidate.object_to_null = null
+      candidate.array_to_object = {}
+      candidate.object_to_array = []
     }
 
-    limitModification(object, ['foo.bar', 'qux'], modifier)
+    limitModification(object, Object.keys(object), modifier)
 
     expect(object).toEqual({
-      foo: { bar: 'bar' },
-      qux: 'qux',
+      string_to_undefined: 'bar',
+      string_to_number: 'qux',
+      null_to_object: null,
+      object_to_null: {},
+      array_to_object: [],
+      object_to_array: {},
```

Comment:
with this strict approach, I am not super confortable with:
- forcing customers to set context to `{}` to clear it, allowing undefined or null to clear the context would seem less surprising to me
- relying on the fact the the commonContext is init with `{}` to allow customers to overwrite the context with an object, if for some reason we change this implementation detail and end up with an undefined context before beforeSend, it would silently break this use case 

wdyt?

Reply:
> with this strict approach, I am not super confortable with:
> 
>     * forcing customers to set context to `{}` to clear it, allowing undefined or null to clear the context would seem less surprising to me

I think this is an edge case, but even then, using an empty object to set an empty context does not seem surprising to me.

> 
>     * relying on the fact the the commonContext is init with `{}` to allow customers to overwrite the context with an object, if for some reason we change this implementation detail and end up with an undefined context before beforeSend, it would silently break this use case

I don't think this is an implementation detail:

* the `context` property *needs* to be defined in the original object, because the object passed to the `beforeSend` callback is expected to have a defined `context` property. Unit tests are already ensuring this, but I will add a proper unit test to make it more explicit.

* `commonContext.context` is, in fact, the "global context", which has to be initialized to an empty object (at least to expose it via `getRumGlobalContext`)

If the `context` ends up undefined somehow, it will break at least a few tests.

Reply:
> I think this is an edge case, but even then, using an empty object to set an empty context does not seem surprising to me.

my point was more about the fact that we can't empty it with undefined or null

> If the context ends up undefined somehow, it will break at least a few tests.

indeed, I'd be a bit more confortable with some higher level tests though